app/services/contributors_generator.rb

```diff
@@ -0,0 +1,243 @@
+# typed: true
+# frozen_string_literal: true
+
+# generates Cocina::Models::Contributors to be used by DescriptionGenerator (ultimately in a Cocina::Models::RequestDRO)
+# rubocop:disable Metrics/ClassLength
+class ContributorsGenerator
+  extend T::Sig
+
+  sig { params(work: Work).returns(T::Array[Cocina::Models::Contributor]) }
+  def self.generate(work:)
+    new(work: work).generate
+  end
+
+  sig { params(work: Work).returns(T::Array[Cocina::Models::DescriptiveValue]) }
+  def self.form_from_contributors(work:)
+    new(work: work).form_from_contributors
+  end
+
+  sig { params(work: Work).void }
+  def initialize(work:)
+    @work = work
+  end
+
+  sig { returns(T::Array[Cocina::Models::Contributor]) }
+  def generate
+    result = []
+
+    work.contributors.each do |work_form_contributor|
+      result << contributor(work_form_contributor)
+    end
+
+    result
+  end
+
+  FORM_EVENT = Cocina::Models::DescriptiveValue.new(
+    {
+      value: 'Event',
+      type: 'resource types',
+      source: { value: 'DataCite resource types' }
+    }
+  )
+
+  sig { returns(T::Array[Cocina::Models::DescriptiveValue]) }
+  def form_from_contributors
+    work.contributors.each do |contributor|
+      next unless contributor.role == 'Conference' || contributor.role == 'Event'
+
+      return [FORM_EVENT]
+    end
+
+    []
+  end
```

Comment:
Let me see if I'm reading this right: if *any* contributor has a role of conference or event, we return a single descriptive value? That's true if there's one conf/event contrib or three or seven, and there's no information from the contributor that is returned? If that's true, I wonder if the `#each` here could be jettisoned in favor of something like:

```ruby
return [FORM_EVENT] if work.contributors.pluck(:role).any? { |role| ['Event', 'Conference'].include?(role) }

[]
```

Reply:
yes ... but is that actually clearer?  Or just more clever?

and correct - only one such value regardless of how many contributors.

Reply:
this works, but the above did not:

```ruby
work.contributors.select { |c| ['Event', 'Conference'].include?(c.role) }.empty?
```